cblox/src/core/tsdf_esdf_submap.cpp

```diff
@@ -47,4 +47,59 @@ bool TsdfEsdfSubmap::saveToStream(std::fstream* outfile_ptr) const {
   return true;
 }
 
+bool TsdfEsdfSubmap::LoadFromStream(const Config& config,
+                                    std::fstream* proto_file_ptr,
+                                    uint64_t* tmp_byte_offset_ptr,
+                                    TsdfEsdfSubmap::Ptr* submap_ptr) {
```

Comment:
Is using a ptr to a smart ptr convention used throughout cblox? In our repos we usually just used smart ptr (or raw ptr, but not ptr to a smart ptr) as an output variable.

Reply:
If I'm correct, we have to do this in this case. The reason is that we want to modify the smart pointer at the caller site. If we were to just pass a smart pointer we would be modifying a copy. This also done in the voxblox loading function [here](https://github.com/ethz-asl/voxblox/blob/master/voxblox/include/voxblox/io/layer_io.h#L47).

If you can see a better way, let me know, because I agree that the syntax here is kind of difficult to interpret and horrible. 

The thing that we really want is a submap as a return value. But without std::optional or exceptions, we're stuck returning a bool. Maybe we could get rid of the flag and return a smart ptr (and a nullptr) in the case the function doesn't work.